src/Chapter3.hs

```diff
@@ -479,7 +511,49 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
+data House
+  = One
+  | Two
+  | Three
+  | Four
+
+houseSize :: House -> Int
+houseSize house = case house of
+  One   -> 1
+  Two   -> 2
+  Three -> 3
+  Four  -> 4
```

Comment:
Wonderful idea to have it as a sum type! This is a very nice way to use the type system to make incorrect states unrepresentable 👏🏼 
It is actually the pattern of FP programming, and you can read more about it here:

* https://kowainik.github.io/posts/haskell-mini-patterns#make-illegal-states-unrepresentable